DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_full_fdb_traffic.py

```diff
@@ -0,0 +1,187 @@
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "master": bridge, "operstate": "up"} for port in ports]}])
+    err_msg = f"Verify that bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await BridgeLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "learning": True} for port in ports]}])
+    err_msg = f"Verify that entities set to learning 'ON' state.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    address_map = (
+        # swp port, tg port,    tg ip,     gw,        plen
+        (ports[0], tg_ports[0], "1.1.1.2", "1.1.1.1", 24),
+        (ports[1], tg_ports[1], "1.1.1.3", "1.1.1.1", 24),
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    streams = {
+        "streamA": {
+            "ip_source": dev_groups[tg_ports[0]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[1]][0]["name"],
+            "srcMac": {"type": "increment",
+                       "start": "00:00:00:00:00:35",
+                       "step": "00:00:00:00:10:00",
+                       "count": 4000},
+            "dstMac": "aa:bb:cc:dd:ee:11",
+            "type": "raw",
+            "protocol": "802.1Q",
+            "rate": "1000",
+        }
+    }
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # check the traffic stats
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Traffic Item Statistics")
+    for row in stats.Rows:
+        loss = tgen_utils_get_loss(row)
+        assert loss == 0, f"Expected loss: 0%, actual: {loss}%"
+
+    rc, out = await dent_dev.run_cmd("bridge fdb show br br0   |  grep 'extern_learn.*offload'  |  wc -l")
+    assert rc == 0, f"Failed to grep 'extern_learn.*offload'.\n"
+
+    amount = int(out) - ixia_vhost_mac_count
+    err_msg = f"Expected count of extern_learn offload entities: 4000, Actual count: {amount}"
+    assert amount  == 4000, err_msg
+
+    out = await BridgeFdb.show(input_data=[{device_host_name: [{"options": "-j"}]}],
+                               parse_output=True)
+    assert out[0][device_host_name]["rc"] == 0, "Failed to get fdb entry.\n"
+
+    fdb_entries = out[0][device_host_name]["parsed_output"]
+    learned_macs = [en["mac"] for en in fdb_entries if "mac" in en]
+    list_macs = ["00:00:00:00:00:35", "00:00:00:08:f0:35"]
+    for mac in list_macs:
+        err_msg = f"Expected MACs are not found in FDB, but found MACs:{out}\n"
+        assert mac in learned_macs, err_msg
```

Comment:
"in" operation means for loop against learned_macs. Since that list is 4k, you go through 4k entries 2 times.
I'm curious whether the first level loop against learned_macs could be faster?....

In any case, I do not understand why do we check only those MACs.